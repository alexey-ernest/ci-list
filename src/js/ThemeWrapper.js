/**
 * Wraps component with theme provider to avoid Material UI errors.
 */

import React, {Component, PropTypes} from 'react';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import {
  lightBlue500,
  lightBlue700,
  grey400,
  pinkA200,
  grey200,
  grey500,
  darkBlack,
  white,
  grey300,
  fullBlack
} from 'material-ui/styles/colors';
import {fade} from 'material-ui/utils/colorManipulator';


// More on Colors: http://www.material-ui.com/#/customization/colors
var muiTheme = getMuiTheme({
  palette: {
    primary1Color: lightBlue500,
    primary2Color: lightBlue700,
    primary3Color: grey400,
    accent1Color: pinkA200,
    accent2Color: grey300,
    accent3Color: grey500,
    textColor: darkBlack,
    alternateTextColor: white,
    canvasColor: white,
    borderColor: grey300,
    disabledColor: fade(darkBlack, 0.3),
    pickerHeaderColor: lightBlue500,
    clockCircleColor: fade(darkBlack, 0.07),
    shadowColor: fullBlack,
  }
});

export default class ThemeWrapper extends Component {

  static childContextTypes = {
    muiTheme: React.PropTypes.object.isRequired
  };

  getChildContext() {
    return {muiTheme: muiTheme};
  }

  render() {
    return (
      <MuiThemeProvider muiTheme={muiTheme}>
        {this.props.children}
      </MuiThemeProvider>
    );
  }
}
