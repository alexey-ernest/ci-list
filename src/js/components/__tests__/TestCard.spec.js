
import React from 'react';
import renderer from 'react-test-renderer';

import ThemeWrapper from '../../ThemeWrapper';
import TestCard from '../TestCard.react';


jest.mock('react-dom');

describe('TestCard component', () => {

  const data = {
    id: 'abcd',
    tests: {
      total: 150,
      passed: 102,
      failed: 48,
      coverage: 33.1
    }
  };

  it('should not render without job.tests provided', () => {
    const component = renderer.create(
      <ThemeWrapper>
        <TestCard job={{}} />
      </ThemeWrapper>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const component = renderer.create(
      <ThemeWrapper>
        <TestCard job={data} />
      </ThemeWrapper>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClick handler when clicking on image', () => {
    const onClick = jasmine.createSpy();

    const component = renderer.create(
      <ThemeWrapper>
        <TestCard job={data} onClick={onClick} />
      </ThemeWrapper>
    );

    let tree = component.toJSON();

    // click on image
    tree.children[0].children[0].children[0].children[0].props.onClick();

    expect(onClick.calls.count()).toBe(1);
  });
});
