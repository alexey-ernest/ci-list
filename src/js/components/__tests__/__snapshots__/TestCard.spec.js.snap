exports[`TestCard component should not render without job.tests provided 1`] = `null`;

exports[`TestCard component should render correctly 1`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": "2px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1
    }
  }>
  <div
    style={
      Object {
        "paddingBottom": 8
      }
    }>
    <div
      style={
        Object {
          "muiPrepared": true,
          "position": "relative"
        }
      }>
      <div
        style={
          Object {
            "muiPrepared": true
          }
        }>
        <img
          className={undefined}
          onClick={[Function anonymous]}
          src="images/crash.png"
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%"
            }
          } />
      </div>
      
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 16,
          "paddingBottom": 0,
          "position": "relative"
        }
      }>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "block",
            "fontSize": 16,
            "fontWeight": 400,
            "lineHeight": "36px",
            "muiPrepared": true
          }
        }>
        Tests
      </span>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "muiPrepared": true
          }
        } />
    </div>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "lineHeight": 1.5,
          "muiPrepared": true,
          "padding": 16,
          "paddingTop": 6
        }
      }>
      Pass rate: 
      68.0
      %, Coverage: 
      33.1
      %
    </div>
  </div>
</div>
`;
